BMSSender/BMSSenderTests/BMSSenderTests.cs

```diff
@@ -0,0 +1,23 @@
+﻿using BMSSender;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace BMSSenderTests
+{
+    [TestClass]
+    public class BMSSenderTests
+    {
+        [TestMethod]
+        public void WhenFetchBatteryParamtersCalled_ThenReturnValidBatteryChargeStateReading()
```

Comment:
These tests are checking random number generation. They may pass even when the ranges are wrong. Focus on testing your functionality - the most crucial is the output format.
In the absence of a test for the output format, any mistakes will only be caught only in integration.